src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -530,6 +537,95 @@ public Result onBuildCompleted(Participation participation) {
         return result;
     }
 
+    @Override
+    public String getPlanKey(Object requestBody) throws BambooException {
+        try {
+            Map<String, Object> requestBodyMap = (Map<String, Object>) requestBody;
+            Map<String, Object> planMap = (Map<String, Object>) requestBodyMap.get("plan");
+            String planKey = (String) planMap.get("key");
+
+            return planKey;
+
+        } catch (Exception e) {
+            log.error("Error when getting plan key");
+            throw new BitbucketException("Could not get plan key", e);
+        }
+    }
+
+    @Override
+    public Result onBuildCompletedNew(Participation participation, Object requestBody) throws Exception {
```

Comment:
Could you add JavaDoc comments here?

Reply:
Not needed as existing in interface JavaDoc.